core/struct/new_spec.rb

```diff
@@ -71,26 +71,49 @@ def obj.to_sym() :ruby end
     klass.instance_variables.should include(:@something_else)
   end
 
-  it "creates a constant in subclass' namespace" do
-    struct = StructClasses::Apple.new('Computer', :size)
-    struct.should == StructClasses::Apple::Computer
+  context "with a block" do
+    it "allows class to be modified via the block" do
+      klass = Struct.new(:version) do
+        def platform
+          :ruby
+        end
+      end
+      instance = klass.new('2.2')
+
+      instance.version.should == '2.2'
+      instance.platform.should == :ruby
+    end
+
+    it "passes same struct class to the block" do
+      given = nil
+      Struct.new(:attr) do |block_parameter|
+        given = block_parameter
+      end.should equal(given)
```

Comment:
My fault, the return value is indeed always the new struct class.
But I think it's still easier to read this way.